src/pathfinding_service/service.py

```diff
@@ -114,162 +106,152 @@ def __init__(
+        # the network addresses here, all else is done later.
+        token_networks_changed = (
+            self.blockchain_state.token_network_addresses
+            != new_chain_state.token_network_addresses
+        )
+        if token_networks_changed:
+            self.blockchain_state.token_network_addresses = new_chain_state.token_network_addresses
+        #     self.context.db.update_state(self.context.ms_state)
+
+        # Now set the updated chain state to the context, will be stored later
+        self.blockchain_state = new_chain_state
+        for event in events:
+            self.handle_channel_event(event)
+
+        self.blockchain_state.latest_known_block = last_block
 
     def stop(self) -> None:
-        self.token_network_registry_listener.stop()
-        for task in self.token_network_listeners:
-            task.stop()
         self.matrix_listener.stop()
         self.is_running.set()
         self.matrix_listener.join()
 
-    def follows_token_network(self, token_network_address: Address) -> bool:
+    def follows_token_network(self, token_network_address: TokenNetworkAddress) -> bool:
         """ Checks if a token network is followed by the pathfinding service. """
-        assert is_checksum_address(token_network_address)
-
         return token_network_address in self.token_networks.keys()
 
-    def _get_token_network(self, token_network_address: Address) -> Optional[TokenNetwork]:
+    def get_token_network(
+        self, token_network_address: TokenNetworkAddress
+    ) -> Optional[TokenNetwork]:
         """ Returns the `TokenNetwork` for the given address or `None` for unknown networks. """
+        return self.token_networks.get(token_network_address)
 
-        assert is_checksum_address(token_network_address)
-        if not self.follows_token_network(token_network_address):
-            return None
-        else:
-            return self.token_networks[token_network_address]
-
-    def handle_channel_event(self, event: Dict) -> None:
-        event_name = event['event']
-
-        if event_name == ChannelEvent.OPENED:
+    def handle_channel_event(self, event: Event) -> None:
+        if isinstance(event, ReceiveTokenNetworkCreatedEvent):
+            self.handle_token_network_created(event)
+        elif isinstance(event, ReceiveChannelOpenedEvent):
             self.handle_channel_opened(event)
-        elif event_name == ChannelEvent.DEPOSIT:
+        elif isinstance(event, ReceiveChannelNewDepositEvent):
             self.handle_channel_new_deposit(event)
-        elif event_name == ChannelEvent.CLOSED:
+        elif isinstance(event, ReceiveChannelClosedEvent):
             self.handle_channel_closed(event)
         else:
             log.debug('Unhandled event', evt=event)
 
-    def handle_channel_opened(self, event: Dict) -> None:
-        token_network = self._get_token_network(event['address'])
+    def handle_token_network_created(self, event: ReceiveTokenNetworkCreatedEvent) -> None:
```

Comment:
This is something we might want to change:
Currently we create the `TokenNetwork` when we receive the according event. However, until the first channel this doesn't make to much sense, so it might be nice to create the network on demand when the first channel is opened.